config/filesystems.php

```diff
@@ -58,8 +58,17 @@
             'key' => env('AWS_ACCESS_KEY_ID'),
             'secret' => env('AWS_SECRET_ACCESS_KEY'),
             'region' => env('AWS_DEFAULT_REGION'),
-            'bucket' => env('AWS_BUCKET'),
+            'bucket' => env('AWS_BUCKET', 'cdash'),
             'url' => env('AWS_URL'),
+            'endpoint' => env('AWS_ENDPOINT'),
+            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
+            // N.B. Laravel downloads files from S3 to a temporary directory on the
+            // local file system before serving them to clients!
+            // Setting 'stream_reads=true' would allow us to stream from S3 directly,
+            // but we can't use it because of our various calls to rewind().
+            // See https://github.com/laravel/framework/discussions/49232
+            // for more details.
+            // 'stream_reads' => true,
```

Comment:
Does Laravel handle cleanup of the files it downloads?

Reply:
Yes. Testing locally I can see the temporary file in `/tmp/` while a large download is in progress, and it disappears afterward. I had trouble tracking down the relevant lines of code in the vendor directory, but here's some supporting docs:

https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_configuration.html#config-http

https://docs.aws.amazon.com/aws-sdk-php/v2/guide/service-s3.html#downloading-objects